cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -473,46 +486,34 @@ struct datetime_formatter {
       return;
     }
 
-    // first, convert to days so we can handle months, leap years, etc.
-    int32_t days = static_cast<int32_t>(timestamp);  // default to days
-    if (units == timestamp_units::hours)
-      days = scale_time(timestamp, 24L);
-    else if (units == timestamp_units::minutes)
-      days = scale_time(timestamp, 1440L);  // 24*60
-    else if (units == timestamp_units::seconds)
-      days = scale_time(timestamp, 86400L);  // 24*60*60
-    else if (units == timestamp_units::ms)
-      days = scale_time(timestamp, 86400000L);
-    else if (units == timestamp_units::us)
-      days = scale_time(timestamp, 86400000000L);
-    else if (units == timestamp_units::ns)
-      days = scale_time(timestamp, 86400000000000L);
-    days = days + 719468;  // 719468 is days between 0000-00-00 and 1970-01-01
-
-    int32_t const daysInEra     = 146097;  // (400*365)+97
-    int32_t const daysInCentury = 36524;   // (100*365) + 24;
-    int32_t const daysIn4Years  = 1461;    // (4*365) + 1;
-    int32_t const daysInYear    = 365;
-
-    // code logic handles leap years in chunks: 400y,100y,4y,1y
-    int32_t year  = 400 * (days / daysInEra);
-    days          = days % daysInEra;
-    int32_t leapy = days / daysInCentury;
-    days          = days % daysInCentury;
-    if (leapy == 4) {  // landed exactly on a leap century
-      days += daysInCentury;
-      --leapy;
-    }
-    year += 100 * leapy;
-    year += 4 * (days / daysIn4Years);
-    days  = days % daysIn4Years;
-    leapy = days / daysInYear;
-    days  = days % daysInYear;
-    if (leapy == 4) {  // landed exactly on a leap year
-      days += daysInYear;
-      --leapy;
-    }
-    year += leapy;
+    // with C++17, user structured bindings and const
```

Comment:
C++17? we use C++14 right?

Reply:
Yes, my comment mean "when we have C++17..."